### PR TITLE
Adds action to Backup Wallet text in backup notification

### PR DESCRIPTION
**Pull request by ryanml**
Fixes: https://github.com/brave/brave-browser/issues/2739
UI: https://github.com/brave/brave-ui/pull/374

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [ ] Windows
  - [ ] macOS
  - [ ] Linux
- [x] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:
- `brave.rewards.backup_notification_frequency` & `brave.rewards.notification_timer_interval`
can be adjusted in your profiles `Preferences` file to shorten the notification frequency and timer interval from 7 days and 15 minutes to something shorter.

1. Enable Rewards, fund wallet
2. Wait for Backup Wallet notification
3. Click the 'Backup Wallet' text
4. Ensure that a tab is opened with `brave://rewards#backup-restore`
5. Ensure that the backup modal is open

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

File: components/brave_rewards/resources/extension/brave_rewards/components/panel.tsx [hunk 1]
```diff
@@ -293,7 +299,11 @@ export class Panel extends React.Component<Props, State> {
     const { balance, rates, grants } = this.props.rewardsPanelData.walletProperties
     const publisher: RewardsExtension.Publisher | undefined = this.getPublisher()
     const converted = utils.convertBalance(balance.toString(), rates)
-    const notification = this.getNotification()
+    const notification: any = this.getNotification()
+    const notificationType = notification.notification.type
```
Comment (NejcZdovc) [inline components/brave_rewards/resources/extension/brave_rewards/components/panel.tsx]: can we move this logic into separate function as I see that we will be adding more of this in the future

Comment (ryanml) [reply components/brave_rewards/resources/extension/brave_rewards/components/panel.tsx]: @NejcZdovc done

Comment (NejcZdovc) [reply components/brave_rewards/resources/extension/brave_rewards/components/panel.tsx]: I still see it in the render. To which function it was moved?

File: browser/ui/webui/brave_webui_source.cc [hunk 2]
```diff
@@ -264,6 +264,8 @@ void CustomizeWebUIHTMLSource(const std::string &name, content::WebUIDataSource*
         { "allowTip", IDS_BRAVE_UI_ALLOW_TIP },
         { "amount", IDS_BRAVE_UI_AMOUNT },
         { "backup", IDS_BRAVE_UI_BACKUP },
+        { "backupNow", IDS_BRAVE_UI_BACKUP_NOW },
```
Comment (NejcZdovc) [inline browser/ui/webui/brave_webui_source.cc]: why is this needed as we only display notification in extensions?

File: components/resources/brave_components_strings.grd [hunk 3]
```diff
@@ -272,6 +272,8 @@
       <message name="IDS_BRAVE_UI_ALLOW_TIP" desc="">Allow tips on</message>
       <message name="IDS_BRAVE_UI_AMOUNT" desc="">Amount</message>
       <message name="IDS_BRAVE_UI_BACKUP" desc="">Backup</message>
+      <message name="IDS_BRAVE_UI_BACKUP_NOW" desc="">Backup Now</message>
```
Comment (NejcZdovc) [inline components/resources/brave_components_strings.grd]: why is this needed as we only display notification in extensions?

File: components/brave_rewards/resources/extension/brave_rewards/components/panel.tsx (hunk 1 above)
Comment (ryanml) [reply components/brave_rewards/resources/extension/brave_rewards/components/panel.tsx]: done, function is here: https://github.com/brave/brave-core/pull/1641/files#diff-4067754668fe25a0ce8194d60d8acf74R220

File: components/resources/brave_components_strings.grd (hunk 3 above)
Comment (ryanml) [reply components/resources/brave_components_strings.grd]: was an oversight removed

File: browser/ui/webui/brave_webui_source.cc (hunk 2 above)
Comment (ryanml) [reply browser/ui/webui/brave_webui_source.cc]: removed

File: browser/ui/webui/brave_webui_source.cc [hunk 4]
```diff
@@ -264,6 +264,7 @@ void CustomizeWebUIHTMLSource(const std::string &name, content::WebUIDataSource*
         { "allowTip", IDS_BRAVE_UI_ALLOW_TIP },
         { "amount", IDS_BRAVE_UI_AMOUNT },
         { "backup", IDS_BRAVE_UI_BACKUP },
+        { "backupWalletTitle", IDS_BRAVE_UI_BACKUP_WALLET_TITLE },
```
Comment (NejcZdovc) [inline browser/ui/webui/brave_webui_source.cc]: do we use this translation in webui?

File: components/resources/brave_components_strings.grd [hunk 5]
```diff
@@ -272,6 +272,7 @@
       <message name="IDS_BRAVE_UI_ALLOW_TIP" desc="">Allow tips on</message>
       <message name="IDS_BRAVE_UI_AMOUNT" desc="">Amount</message>
       <message name="IDS_BRAVE_UI_BACKUP" desc="">Backup</message>
+      <message name="IDS_BRAVE_UI_BACKUP_WALLET_TITLE" desc="">Backup Wallet</message>
```
Comment (NejcZdovc) [inline components/resources/brave_components_strings.grd]: same as above

Comment (ryanml) [reply components/resources/brave_components_strings.grd]: done

File: browser/ui/webui/brave_webui_source.cc (hunk 4 above)
Comment (ryanml) [reply browser/ui/webui/brave_webui_source.cc]: nope, removed

File: components/brave_rewards/resources/extension/brave_rewards/_locales/en_US/messages.json [hunk 6]
```diff
@@ -185,6 +185,10 @@
     "message": "You have a grant waiting for you.",
     "description": "Description for new grant notification"
   },
+  "backupNow": {
```
Comment (NejcZdovc) [inline components/brave_rewards/resources/extension/brave_rewards/_locales/en_US/messages.json]: you need to add `backupNow` to `locale_api.ts` as well. Current view
image

File: components/brave_rewards/resources/extension/brave_rewards/components/panel.tsx [hunk 7]
```diff
@@ -115,6 +115,12 @@ export class Panel extends React.Component<Props, State> {
     this.actions.getGrantCaptcha()
   }
 
+  onBackupWallet = () => {
+    chrome.tabs.create({
+      url: 'chrome://rewards#backup-restore'
+    })
```
Comment (NejcZdovc) [inline components/brave_rewards/resources/extension/brave_rewards/components/panel.tsx]: We should clear notification when user clicks backup now

File: components/brave_rewards/resources/extension/brave_rewards/_locales/en_US/messages.json (hunk 6 above)
Comment (ryanml) [reply components/brave_rewards/resources/extension/brave_rewards/_locales/en_US/messages.json]: done

File: components/brave_rewards/resources/extension/brave_rewards/components/panel.tsx (hunk 7 above)
Comment (ryanml) [reply components/brave_rewards/resources/extension/brave_rewards/components/panel.tsx]: done

**Comment by ryanml** [comment]
@NejcZdovc this one is ready (UI updated to master)
